06-aks-cluster.md

```diff
@@ -4,134 +4,133 @@ Now that the [cluster prequisites and shared Azure service instances are provisi
-      >
-      > - AKS cluster creation
-      > - Flux deployment
-
-   1. Once the GitHub Workflow validation finished successfully, please proceed by merging this PR into `main`.
-
-      > :book: The DevOps team monitors this Workflow execution instance. In this instance it will impact a critical piece of infrastructure as well as the management. This flow works for both new or an existing AKS cluster.
-
-   1. :fast_forward: The cluster is placed under GitOps managed as part of these GitHub Workflow steps. Therefore, you should proceed straight to [Workflow Prerequisites](./07-workload-prerequisites.md).
+1.  Obtain shared services resource details
+
+    ```bash
+    LOGANALYTICSWORKSPACEID=$(az deployment group show -g rg-bu0001a0042-shared -n shared-svcs-stamp --query properties.outputs.logAnalyticsWorkspaceId.value -o tsv)
+    CONTAINERREGISTRYID=$(az deployment group show -g rg-bu0001a0042-shared -n shared-svcs-stamp --query properties.outputs.containerRegistryId.value -o tsv)
+    ACRPRIVATEDNSZONESID=$(az deployment group show -g rg-bu0001a0042-shared -n shared-svcs-stamp --query properties.outputs.acrPrivateDnsZonesId.value -o tsv)
+    ```
+
+1.  Get the corresponding AKS cluster spoke VNet resource IDs for the app team working on the application A0042.
+
+    > :book: The app team will be deploying to a spoke VNet, that was already provisioned by the network team.
+
+    ```bash
+    RESOURCEID_VNET_BU0001A0042_03=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-03 --query properties.outputs.clusterVnetResourceId.value -o tsv)
+    RESOURCEID_VNET_BU0001A0042_04=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-04 --query properties.outputs.clusterVnetResourceId.value -o tsv)
+    ```
+
+    **Automated deploy using GitHub Actions (fork is required)**
+
+    1.  Create the Azure Credentials for the GitHub CD workflow.
+
+        ```bash
+        # Create an Azure Service Principal
+        az ad sp create-for-rbac --name "github-workflow-aks-cluster" --sdk-auth --skip-assignment > sp.json
+        APP_ID=$(grep -oP '(?<="clientId": ").*?[^\\](?=",)' sp.json)
+
+        # Wait for propagation
+        until az ad sp show --id ${APP_ID} &> /dev/null ; do echo "Waiting for Azure AD propagation" && sleep 5; done
+
+        # Assign built-in Contributor RBAC role for creating resource groups and performing deployments at subscription level
+        az role assignment create --assignee $APP_ID --role 'Contributor'
+
+        # Assign built-in User Access Administrator RBAC role since granting RBAC access to other resources during the cluster creation will be required at subscription level (e.g. AKS-managed Internal Load Balancer, ACR, Managed Identities, etc.)
+        az role assignment create --assignee $APP_ID --role 'User Access Administrator'
+        ```
+
+    1.  Create `AZURE_CREDENTIALS` secret in your GitHub repository. For more
```

Comment:
please lets invest a bit now this is our main exp on how adding secrets in a more scripty way.

Reply:
Let me do in the next cycle after take a look to the regulated app

Reply:
It created a task to do this.